WordPress/Classes/Utility/DisplayableImageHelper.m

```diff
@@ -0,0 +1,184 @@
+#import "DisplayableImageHelper.h"
+
+static const NSInteger FeaturedImageMinimumWidth = 640;
+
+@implementation DisplayableImageHelper
+
+/**
+ Get the url path of the image to display for a post.
+
+ @param dict A dictionary representing a posts attachments from the REST API.
+ @return The url path for the featured image or nil
+ */
++ (NSString *)searchPostAttachmentsForImageToDisplay:(NSDictionary *)attachmentsDict
+{
+    NSArray *attachments = [[attachmentsDict dictionaryForKey:@"attachments"] allValues];
+    if ([attachments count] == 0) {
+        return nil;
+    }
+
+    NSString *imageToDisplay;
+
+    attachments = [self filteredAttachmentsArray:attachments];
+
+    NSDictionary *attachment = [attachments firstObject];
+    NSInteger width = [[attachment numberForKey:@"width"] integerValue];
+    if (width >= FeaturedImageMinimumWidth) {
+        imageToDisplay = [attachment stringForKey:@"URL"];
+    }
+
+    return imageToDisplay;
+}
+
++ (NSArray *)filteredAttachmentsArray:(NSArray *)attachments
+{
+    NSString *key = @"mime_type";
+    NSPredicate *predicate = [NSPredicate predicateWithFormat:@"%K BEGINSWITH %@", key, @"image"];
+    attachments = [attachments filteredArrayUsingPredicate:predicate];
+
+    return [attachments sortedArrayUsingComparator:^NSComparisonResult(id obj1, id obj2) {
```

Comment:
This is one of those cases where the new method is not really going to change to code, but simply improve its readability and make the code more atomic.

So, if we had...

```objective-c
return [self sortedAttachmentsArray:attachments];
```

...instead of...

```objective-c
return [attachments sortedArrayUsingComparator:^NSComparisonResult(NSDictionary *attachmentA, NSDictionary *attachmentB) {
    NSString *key = AttachmentsDictionaryKeyWidth;
    NSNumber *widthA = [attachmentA numberForKey:key] ?: @(0);
    NSNumber *widthB = [attachmentB numberForKey:key] ?: @(0);

    if ([widthA integerValue] < [widthB integerValue]) {
        return NSOrderedDescending;
    } else if ([widthA integerValue] > [widthB integerValue]) {
        return NSOrderedAscending;
    } else {
        return NSOrderedSame;
    }
}];
```

...we would gain:

- Readability: the single call line conveys its own meaning to the reader, and allows him to stay focused on the higher level logic.
- Abstraction: no need to check the sorting details unless the sorting breaks.
- More atomic.
- Easier to self document, and contrast with the explicit documentation: by having higher level steps in the methods, it's easier to make the code self documenting and compare the code with the method comments.  For instance one thing that I noticed here is that `+ (NSArray *)filteredAttachmentsArray:(NSArray *)attachments` doesn't hint to the caller that sorting takes place.